### PR TITLE
Fix StaticCallConverter

### PR DESCRIPTION
**Pull request by lifinsky**
## Why is this change proposed?
Fixes [issue #532](https://github.com/ecotoneframework/ecotone-dev/issues/532).

## Description of Changes
The bug occurred in StaticCallConverter, where a converter parameter typed as string was incorrectly treated as compatible with Stringable. This behavior is not allowed in strict mode, since Stringable objects cannot be implicitly cast to string.

This fix ensures that type matching is handled correctly when comparing an object type against a non-object type.

## Pull Request Contribution Terms

- [X] I have read and agree to the contribution terms outlined in [CONTRIBUTING](https://github.com/ecotoneframework/ecotone-dev/blob/main/CONTRIBUTING.md).

### CODE REVIEW

File: packages/Ecotone/src/Messaging/Conversion/StaticCallConverter.php [hunk 1]
```diff
@@ -28,9 +29,21 @@ public function convert($source, Type $sourceType, MediaType $sourceMediaType, T
...
+            return false;
+        }
+
+        if ($sourceType instanceof ObjectType && ! $this->sourceType instanceof ObjectType) {
+            return false;
+        }
+
+        if ($this->targetType instanceof ObjectType && ! $targetType instanceof ObjectType) {
+            return false;
+        }
```
Comment (dgafka) [inline packages/Ecotone/src/Messaging/Conversion/StaticCallConverter.php:44]: Are those two if statement meant to handle not passing Stringable to string and vice versa? 

We also handle union types as UnionType, therefore I don't think instanceof usage will be safe here.

Comment (dgafka) [reply packages/Ecotone/src/Messaging/Conversion/StaticCallConverter.php:44]: If strict mode doesn't allow for conversion between stringable and string, then I don't think we should allow for defining equality between those in Ecotone's type system.

Therefore we could drop representation of stringable as string here:
image

@jlabedo wdyt?

Comment (dgafka) [reply packages/Ecotone/src/Messaging/Conversion/StaticCallConverter.php:44]: If strict mode doesn't allow for conversion between stringable and string, then I don't think we should allow for defining equality between those in Ecotone's type system.

Therefore we could drop representation of stringable as string here:
image

@jlabedo wdyt?

Comment (lifinsky) [reply packages/Ecotone/src/Messaging/Conversion/StaticCallConverter.php:44]: @dgafka Yes, I was thinking about that too.
But the `BuiltinType` looks like a class that’s too generic to contain fixes for the acceptType and accepts methods.

I want to point out that this strict typing issue would also occur with float/integer and string/scalar types.

Comment (lifinsky) [reply packages/Ecotone/src/Messaging/Conversion/StaticCallConverter.php:44]: I’d also like to note that intersection types are not supported by the current type system, which could be critical for converter target types.

Comment (lifinsky) [reply packages/Ecotone/src/Messaging/Conversion/StaticCallConverter.php:48]: @dgafka Updated, I suggest staying on this version until there is more time to add tests for conversion via StaticCallConverter and cover all edge cases.

Comment (jlabedo) [reply packages/Ecotone/src/Messaging/Conversion/StaticCallConverter.php:48]: Hey, the static converter is just a converter that do not need to be registered in the dependency container. It should have the exact same behaviour as `ReferenceServiceConverter` regarding `match` method. The fix is in #536
